File: app/src/main/java/com/keelim/cnubus/mainfragment/SettingFragment.java
package com.keelim.cnubus.mainfragment;

import android.content.Intent;
import android.os.Bundle;

import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.keelim.cnubus.R;
import com.keelim.cnubus.activities.developer.DeveloperActivity;

class SettingFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.settings_preferences);
    }

    @Override
    public boolean onPreferenceTreeClick(Preference preference) { //preferebce 클릭 리스너
        String key = preference.getKey();
        if (key.equals("developer")) {
            Intent intent_developer = new Intent(getContext(), DeveloperActivity.class);
            startActivity(intent_developer);
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/keelim/nandadiagnosis/view/setting/SettingFragment.kt
package com.keelim.nandadiagnosis.view.setting

import android.content.Intent
import android.os.Bundle
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.keelim.nandadiagnosis.R
import com.keelim.nandadiagnosis.view.LoginActivity
import com.keelim.nandadiagnosis.view.WebViewActivity

class SettingFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.settings_preferences)
    }

    override fun onPreferenceTreeClick(preference: Preference): Boolean {
        return when (preference.key) {
            "nandaHome" -> {
                Intent(context, WebViewActivity::class.java).apply {
                    putExtra("URL", "https://keelim.github.io/nandaDiagnosis/")
                    startActivity(this)
                }
                true
            }
            "opensource" -> {
                startActivity(Intent(context, OpenSourceActivity::class.java))
                return true
            }
            "please" -> {
                startActivity(Intent(context, PleaseActivity::class.java))
                return true
            }
            "lab" -> {
                startActivity(Intent(context, LoginActivity::class.java))
                true
            }
            else -> false
        }
    }

}